docs/frontend-system/building-plugins/testing.md

```diff
@@ -0,0 +1,237 @@
+
+Use the `renderInTestApp` helper to render a given component inside a Backstage test app:
+
+```tsx
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { renderInTestApp } from '@backstage/frontend-test-utils';
+import { EntityDetails } from './plugin';
+
+describe('Entity details component', () => {
+  it('should render the entity name and owner', async () => {
+    renderInTestApp(<EntityDetails owner="tools" name="test" />);
+
+    await expect(
+      screen.getByText('The entity "test" is owned by "tools"'),
+    ).resolves.toBeInTheDocument();
+  });
+});
+```
+
+It's important to highlight that mocking APIs for components is different from mocking them for extensions. In the snippet below, we wrapped the component within a `TestApiProvider` for mocking the Catalog API:
+
+```tsx
+import React from 'react';
+import { screen } from '@testing-library/react';
+import {
+  renderInTestApp,
+  TestApiProvider,
+} from '@backstage/frontend-test-utils';
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { CatalogApi, catalogApiRef } from '@backstage/plugin-catalog-react';
+import { EntityDetails } from './plugin';
+
+describe('Entity details component', () => {
+  it('should render the entity name and owner', async () => {
+    const catalogApiMock: Partial<CatalogApi> = {
+      getEntityFacets: () =>
+        Promise.resolve({
+          facets: {
+            'relations.ownedBy': [{ count: 1, value: 'group:default/tools' }],
+          },
+        }),
+    };
+
+    const entityRef = stringifyEntityRef({
+      kind: 'Component',
+      namespace: 'default',
+      name: 'test',
+    });
+
+    renderInTestApp(
+      <TestApiProvider apis={[[catalogApiRef, catalogApiMock]]}>
+        <EntityDetails entitRef={entityRef} />
+      </TestApiProvider>,
+    );
+
+    await expect(
+      screen.getByText('The entity "test" is owned by "tools"'),
+    ).resolves.toBeInTheDocument();
+  });
+});
+```
+
+## Testing features
+
+To facilitate testing of frontend features, the `@backstage/frontend-test-utils` package provides a tester class which starts up an entire frontend harness, complete with a number of default features. You can then provide overrides for extensions whose behavior you need to adjust for the test run.
+
+A number of features (frontend extensions and overrides) are also accepted by the tester. Here are some examples of how these facilities can be useful:
+
+### Single extension
+
+In order to test an extension in isolation, you simply need to pass it into the tester factory, then call the render method on the returned instance:
+
+```tsx
+import { screen } from '@testing-library/react';
+import { createExtensionTester } from '@backstage/frontend-test-utils';
+import { indexPageExtension } from './plugin';
+
+describe('Index page', () => {
+  it('should render a the index page', () => {
+    const tester = createExtensionTester(indexPageExtension);
+
+    tester.render();
```

Comment:
```suggestion
    createExtensionTester(indexPageExtension).render();
```

Pending other possible versions of this API I think we should encourage the direct chain version of calling it for now. Same for the examples below

Reply:
Done 🙌🏻